mlrun/api/db/sqldb/db.py

```diff
@@ -984,6 +984,30 @@ def store_function(
         self.tag_objects_v2(session, [fn], project, tag)
         return hash_key
 
+    def get_function(self, session, name, project="", tag="", hash_key="") -> dict:
+        """
+        In version 1.4.0 we added a normalization to the function name before storing.
+        To be backwards compatible and allow users to query old non-normalized functions,
+        we're providing a fallback to get_function:
+        normalize the requested name and try to retrieve it from the database.
+        If no answer is received, we will check to see if the original name contained underscores,
+        if so, the retrieval will be repeated and the result (if it exists) returned.
+        """
+        normalized_function_name = mlrun.utils.normalize_name(name)
+        try:
+            return self._get_function(
+                session, normalized_function_name, project, tag, hash_key
+            )
+        except mlrun.errors.MLRunNotFoundError as exc:
+            if "_" in name:
+                logger.debug(
+                    f"Function {normalized_function_name} was not found,"
+                    f" trying to find function by its original name"
+                )
```

Comment:
Id change the debug verbose to warning
and change the log to use kwargs, so once we have json formatter, those kwargs would be indexed well

```python
logger.warning("Failed to get underscore-named function, trying without normalization", function_name=name)
```